src/argtable3.c

```diff
@@ -504,11 +504,11 @@ static void arg_cat(char** pdest, const char* src, size_t* pndest) {
     char* end = dest + *pndest;
 
     /*locate null terminator of dest string */
-    while (dest < end && *dest != 0)
+    while (dest < end-1 && *dest != 0)
         dest++;
 
     /* concat src string to dest string */
-    while (dest < end && *src != 0)
+    while (dest < end-1 && *src != 0)
```

Comment:
`end` is past the end of the buffer. `end = dest + buffer_size.`

i.e. if `end` is 0x05, the last time would should enter the loop (& increment `dest`) is when `dest` is 0x03.

**last iteration:** 0x03 < 0x04 -> dest will become 0x04